### ISSUE TITLE
prevQueue doesn't get saved when the room gets unloaded and the current queue is empty

### ISSUE THREAD

**Issue by dyc3** [dyc3/opentogethertube#1365]
### Expected Behavior

When the room gets unloaded and the current queue is empty, the current queue should become the previous queue. In other words, when the current queue is empty, the previous queue should be saved as empty upon room unload.

### Steps To Reproduce

1. create a permanent room
2. add videos
3. unload room so prev queue gets saved
4. load room
5. see previous queue restored (referred to as queue A)
6. remove all videos from queue, including currentSource
7. unload room to try to get prev queue saved
8. load room
9. see queue A is restored when it should be empty

### Environment

- [X] This happens on the official site, opentogethertube.com
- [ ] This happens using a self-hosted version.
- [ ] I'm using the docker image.

**Comment by WuTofu** [comment]
@dyc3, would you mind reviewing the following test I added on one of my branch and see if my understanding of the expected behavior is correct?

```ts
		it("should preserve empty prevQueue on unload and reload when current queue is empty", async () => {
			const roomName = "test-prevqueue-empty";
			try {
				await roommanager.createRoom({ name: roomName, isTemporary: false });
				const room = (await roommanager.getRoom(roomName)).unwrap();

				room.prevQueue = [{ service: "direct", id: "foo" }];
				room.queue = new VideoQueue([]);
				room.currentSource = null;

				await room.sync();

				let dbRoom = await DbRoom.findOne({ where: { name: roomName } });
				expect(dbRoom).not.toBeNull();
				expect(dbRoom?.prevQueue).toEqual([{ service: "direct", id: "foo" }]);

				await roommanager.unloadRoom(roomName, UnloadReason.Keepalive);

				dbRoom = await DbRoom.findOne({ where: { name: roomName } });
				expect(dbRoom).not.toBeNull();
				expect(dbRoom?.prevQueue).toBeNull();

				const loaded = (await roommanager.getRoom(roomName)).unwrap();
				expect(loaded.prevQueue).toBeNull();
			} finally {
				await DbRoom.destroy({ where: { name: roomName } });
			}
		});
```

**Comment by dyc3** [comment]
its a bit hard to follow the flow, but yeah i think so?
